vortex-flatbuffers/flatbuffers/vortex-serde/message.fbs

```diff
@@ -9,8 +9,28 @@ enum Compression: uint8 {
     None = 0,
 }
 
+/// An ArrayData describes the hierarchy of an array as well as the locations of the data buffers that appear
+/// immediately after the message in the byte stream.
+table ArrayMessage {
+    /// The array's hierarchical definition.
+    array: Array;
+    /// The row count of the array.
+    row_count: uint64;
+    /// The locations of the data buffers of the array, in ascending order of offset.
+    buffers: [Buffer];
+}
+
+table Buffer {
+    /// The length of the buffer in bytes.
+    length: uint64;
+    /// The length of any padding bytes written immediately following the buffer.
+    padding: uint16;
+    /// The minimum alignment of the buffer.
+    alignment: uint16;
+}
+
 union MessageHeader {
-    ArrayData,
+    ArrayMessage,
```

Comment:
yeah, there's a weird lack of symmetry in naming here: `ArrayMessage`, `Buffer`, `DType`